lib/drivers/external_screen.c

```diff
@@ -123,4 +124,14 @@ void external_screen_static_with_scroll(const char *static_text, const char *scr
         }
     }
 }
+
+void external_screen_update_time_from_clock()
```

Comment:
can this method be please moved to display_controller. The drivers should be only operating the hardware, not handling logic such as getting the values from other files.